src/crypto/clu_decrypt.c

```diff
@@ -0,0 +1,223 @@
+                    ret = ENCRYPT_ERROR;
+                }
+            }
+            else if (ret == 0 && keyType == 2) {
+                for (i = 0; i < size; i++) {
+
+                    /* ensure key is set */
+                    if (key[i] == 0 || key[i] == '\0') {
+                        continue;
+                    }
+                    else {
+                        keyVerify++;
+                    }
+                }
+                if (keyVerify == 0) {
+                    WOLFCLU_LOG(WOLFCLU_L0, "the key is all zero's or not set.");
+                    ret = ENCRYPT_ERROR;
+                }
+            }
+        }
+
+        /* Read in 1kB */
+        if (ret == 0 &&
+                (ret = (int)XFREAD(input, 1, MAX_LEN, inFile)) != MAX_LEN) {
+            if (feof(inFile)) {
+                tempMax = ret;
+                ret = 0; /* success */
+            }
+            else {
+                WOLFCLU_LOG(WOLFCLU_L0, "Input file does not exist.");
+                ret = FREAD_ERROR;
+            }
+        }
+
+        /* sets pwdKey decrypts the message to output from input length */
+#ifdef HAVE_CAMELLIA
+        if (ret == 0 &&
+                (alg == WOLFCLU_CAMELLIA128CBC ||
+                 alg == WOLFCLU_CAMELLIA192CBC ||
+                 alg == WOLFCLU_CAMELLIA256CBC)) {
+            ret = wc_CamelliaSetKey(&camellia, key, block, iv);
+            if (ret == 0) {
+                wc_CamelliaCbcDecrypt(&camellia, output, input, tempMax);
+            }
+        }
+#endif
+        if (ret == 0 && currLoopFlag == lastLoopFlag) {
+            if (output != NULL && salt[0] != 0) {
+                /* reduces length based on number of padded elements  */
+                pad = output[tempMax-1];
+                /* adjust length for padded bytes and salt size */
+                length -= pad + sbSize;
+                if (length < 0) {
+                    WOLFCLU_LOG(WOLFCLU_L0, "bad length %d found", length);
+                    ret = -1;
+                    break;
+                }
+                /* reset tempMax for smaller decryption */
+                XFWRITE(output, 1, length, outFile);
+                break;
+            }
+            else {
+                if (output != NULL)
+                    XFWRITE(output, 1, tempMax, outFile);
+
+                XMEMSET(input, 0, tempMax);
+                if (output != NULL)
+                    XMEMSET(output, 0, tempMax);
+                break;
+            }
+        }
+        /* writes output to the outFile */
+        if (ret == 0 && output != NULL)
+            XFWRITE(output, 1, tempMax, outFile);
+
+        if (ret == 0) {
+            XMEMSET(input, 0, tempMax);
+            if (output != NULL)
+                XMEMSET(output, 0, tempMax);
+
+            currLoopFlag++;
+            length -= tempMax;
+        }
+    }
+    /* closes the opened files and frees memory */
+    XMEMSET(input, 0, MAX_LEN);
```

Comment:
wolfCLU does this a lot, zeroing out buffers before freeing them. Why? I can see the logic if the buffer contains sensitive data, but when it doesn't, we shouldn't waste the cycles / add unneeded complexity to the code.